### DISCUSSION TITLE
.Net Blazor Hybrid App has a weird System.NullReferenecException

### DISCUSSION THREAD

**Discussion by cyrilgupta** [dotnet/maui#19278] [Q&A]
0

I have a Blazor Hybrid app (.Net Maui) for Android and I am facing a weird issue that I can't pin at all even after trying a lot. It seems I am missing something fundamental and basic.

A lot of my users are getting weird ANRs, and logging through Sentry, all I am getting is a NullReference error in OnParametersSet.

The interesting thing is, I am running a Null reference check just one line above. So I am absolutely clueless how this could be happening.

The object is a Global object in a static class. Could that be a problem somehow? I am using a Global Static class to hold many variables which have a lifetime validity throughout the app.
Here's the offending code

    protected override void OnParametersSet()
        {
            try
            {
                TestNull();
    
                Calculate();
            }
            catch(Exception ex)
            {
                LogError(ex);
            }
        }
    
        private void Calculate()
        {
            Name = Globals.Person.Name; //The error is right here.
    
            var dbirth = Globals.Person.DateOfBirth;
            TimeOfBirth = dbirth.ToString("hh:mm:ss tt");
            City = Globals.Person.City;
        }
    }
Has anyone got any ideas about why this could be happening?

**Comment by Xyncgas** [comment]
Either Global.Person.Name is null or MAUI.Blazor is bugged
I usually add int a = 0; after that line and set a break point and hover on Global.Person after breakpoint hit
